internal/controllers/parser.go

```diff
@@ -149,6 +149,54 @@ func UpdateConfigFromAPIOpts(
 				},
 				)
 			}
+			if finalOpts.Auth != nil {
+				upstreamServiceHost := finalOpts.Auth.AuthUpstream.Host.Hostname
+				upstreamServicePort := finalOpts.Auth.AuthUpstream.Host.Port
+
+				clusterName := generateClusterName(upstreamServiceHost, upstreamServicePort)
+
+				var authHeaders []*envoy_config_core_v3.HeaderValue
+				if finalOpts.Auth.Scheme == "cloudentity" {
+					clBuilder.AddAPI(upstreamServiceHost, upstreamServicePort, name, name, routePath, method)
+					authHeaders = []*envoy_config_core_v3.HeaderValue{
+						{
+							Key:   cloudentity.HeaderAuthorizerURL,
+							Value: fmt.Sprintf("https://%s:%d", upstreamServiceHost, upstreamServicePort),
+						},
+						{
+							Key:   cloudentity.HeaderAPIGroup,
+							Value: name,
+						},
+					}
+					upstreamServiceHost = cloudEntityHostname
+					upstreamServicePort = cloudEntityPort
+				}
```

Comment:
Shouldn't there be a `clusterName := generateClusterName(upstreamServiceHost, upstreamServicePort)` like the lines above it because you've just modified `upstreamServiceHost` and `upstreamServicePort`?

Also, I can't work out why you've hard-coded `cloudEntityHostname` and `cloudEntityPort`.